### ISSUE TITLE
Update Tap-Hold QMK settings

### ISSUE THREAD

**Issue by refact0r** [vial-kb/vial-gui#180]
QMK has updated some of the Tap-Hold settings. 
https://github.com/qmk/qmk_firmware/blob/master/docs/tap_hold.md

There is a new Hold on Other Key Press option:
https://github.com/qmk/qmk_firmware/commit/610035dce8f4f7010779e61e1cb590eebc8cea46

Tapping Force Hold has been replaced with Quick Tap Term:
https://github.com/qmk/qmk_firmware/commit/cbabc8dbe6a8476d3082e8bc649d330f87e7b904

**Comment by arrowtype** [comment]
I am newly using a 30% keyboard, and wow, I really need "hold on other key press" as an option to make the Z-shift work as my muscle memory demands. Just commenting to say that I would love for this option to be in Vial! (Overall, I am loving the software, btw – thank you!)

**Comment by Malefaro** [comment]
I would also really like `Hold on Other Key Press` feature. For example, to normally bind `esc` as `MO(..)` to another layer.

**Comment by LyraelRayne** [comment]
What actually seems to be the case is that `IGNORE_MOD_TAP_INTERRUPT` was replaced with `HOLD_ON_OTHER_KEY_PRESS`. It appears Vial implemented this by implementing `get_hold_on_other_key_press` as the negation of the old `get_ignore_mod_tap_interrupt` (i.e. `!(QS.tapping & 2)`) instead of inverting the meaning of the flag in Vial and re-wording it in Vial GUI. So `Hold on Other Key Press` *is* actually present in Vial, it just has its value inverted and the wrong label.

**Comment by andreikonsen** [comment]
> I would also really like `Hold on Other Key Press` feature. For example, to normally bind `esc` as `MO(..)` to another layer.
> 
> **UPD:** Just realized that there is special code LT at layers tab. With that I can easly setup esc on tap -> change layer on hold without tap-dance tapping term issues!

Thanks for mentioning this! The LT(layer) does indeed seem to work as "Hold on Other Key Press."  Using this instead of tapdance just doubled my one-handed typing WPM and reduced errors with half-QWERTY!

**Comment by refact0r** [comment]
> So Hold on Other Key Press is actually present in Vial, it just has its value inverted and the wrong label.
Yeah, at least for the layer/mod tap keys, it seems like Hold On Other Key Press works when Ignore Mod Tap Interrupt is disabled, and does not work when it is enabled.
I think the Tapping Force Hold setting is also inverted in Vial, it only seems to work when I uncheck the setting.

**Comment by gunnar-bell** [comment]
I'm having some trouble with this. It's a rather old post but fingers crossed someone can help?

I have a 60% keyboard, and I have right shift as a tap dance key. Tap for Up, Hold for Right Shift. I'm trying to type a bang (!), but no matter which combination of qmk settings i do, the right shift registers a tap unless i wait the full tapping term.

Has anyone solved this issue?

**Comment by gunnar-bell** [comment]
After reading this thread a few more times it finally clicked that this is solvable via Quantum Keys. Here's a link in the vial docs that also touches on this: https://get.vial.today/manual/mod-tap-tapdance.html. Thanks to the folks above!

**Comment by Padrecc** [comment]
> After reading this thread a few more times it finally clicked that this is solvable via Quantum Keys. Here's a link in the vial docs that also touches on this: https://get.vial.today/manual/mod-tap-tapdance.html. Thanks to the folks above!
> 
> FWIW I also think the QMK settings seem a little strange in vial as noted in the above comments. Still a good issue.

Could you please clarify how it can be solved with quantum keys?

**Comment by gunnar-bell** [comment]
> Could you please clarify how it can be solved with quantum keys?

Sure -- here is a screenshot of my setup:

![Screenshot 2025-01-15 at 1 37 33 PM](https://github.com/user-attachments/assets/7e02af2e-8853-4c7f-a5e8-0962b4ed78ee)

In vial, you click the key you want to map in top pane, then select the Quantum tab in the bottom pan, then select the key you want to map.

The process for mapping the right shift key to be up when tapping for instance would be:
1. Select the Right Shift key in the top pane
2. Select the Quantum tab in the bottom pane
3. Select the RSft_T quantum key
4. Select the lighter gray box in the Right Shift key in the top pane
5. Select the Up key in the bottom pane

You can then repeat similar steps for other combinations of keys. I hope this helps!

**Comment by Padrecc** [comment]
> > Could you please clarify how it can be solved with quantum keys?
> 
> Sure -- here is a screenshot of my setup:
> 
> ![Screenshot 2025-01-15 at 1 37 33 PM](https://github.com/user-attachments/assets/7e02af2e-8853-4c7f-a5e8-0962b4ed78ee)
> 
> In vial, you click the key you want to map in top pane, then select the Quantum tab in the bottom pan, then select the key you want to map.
> 
> The process for mapping the right shift key to be up when tapping for instance would be:
> 
> 1. Select the Right Shift key in the top pane
> 2. Select the Quantum tab in the bottom pane
> 3. Select the RSft_T quantum key
> 4. Select the lighter gray box in the Right Shift key in the top pane
> 5. Select the Up key in the bottom pane
> 
> You can then repeat similar steps for other combinations of keys. I hope this helps!

This is not the issues discussed here and it doesn't address it at all. Let's imagine you have shift key configured exactly same way as in your goide through quantum keys and mapped to F key. Now you need to type "f?'. When you typing fast you tal f for f symbol then immediately tap and hold it for shift for ? symbol. 

Try it and you will see what is wrong. Instead of f? , you will type ffffffff or ff/ 
Quick tap term in qmk can help here but it is not supported by vial.

**Comment by gunnar-bell** [comment]
@Padrecc thanks for letting me know this didn't help you. As I said in my original post, i agree that the qmk settings seem broken or out of date in vial. Hopefully someone can fix!

**Comment by fluxxone** [comment]
Hi, I'm new to Vial and I just built a custom keyboard and I'm having issues with this feature - it doesn't work for me at all. Whether I set the option or not, it doesn't change anything for me. App doesn't but even compiling it in also doesn't. I even tried disabling/enabling the HOLD_ON_OTHER_KEY_PRESS_PER_KEY. Any help would be appreciated.

**Comment by 3v1n0** [comment]
The `HOLD_ON_OTHER_KEY_PRESS` thingy  is not applying to tap-dance keys, right?
